frontend/src/lib/auth.js

```diff
@@ -25,19 +25,21 @@ export function setAuth(token, email) {
 }
 
 export function initializeAuthLib() {
-  if (window.gapi.auth2) {
-    return;
-  }
-
-  window.gapi.load('auth2', () => {
-    window.gapi.auth2.init({
-      client_id: GOOGLE_CLIENT_ID,
-      fetch_basic_profile: false,
-      scope: 'email',
-    }).then(
-      () => this.setState({ready: true})
-    );
-  });
+  return new Promise((resolve, reject) => {
+    if (window.gapi.auth2) {
+      resolve();
+    } else {
+      window.gapi.load('auth2', () => {
+        window.gapi.auth2.init({
+          client_id: GOOGLE_CLIENT_ID,
```

Comment:
this doesn't have to be a secret thing?

Reply:
no, that's just the ID.  the secret is only stored server-side in the secrets.cfg file.

the front-end is responsible for all of the UI / login redirects.  it just gets a temp access token, passes it to the backend, who then returns an API auth token that the frontend stores in localStorage.

I'll eventually add a way for people to log out which will just clear localStorage.